hy/lex/parser.py

```diff
@@ -29,6 +29,16 @@ def mangle(s):
     """Stringify the argument and convert it to a valid Python identifier
     according to Hy's mangling rules."""
 
+    def convert_to_ucs4_list(ustr):
+        if UCS4:
+            return ustr
+        ucs4_list = list(ustr)
+        for i, u in enumerate(ucs4_list):
+            if 0xDC00 > ord(u) > 0xD7FF:
+                ucs4_list[i] += ucs4_list[i + 1]
+                del ucs4_list[i + 1]
```

Comment:
Mutating a list while iterating over it fields weird to me. Using a generator here might be cleaner, but I suspect this would be faster.

Reply:
@vodik - Actually I guess `del` operation is quite expensive - I wrote like that for pythonic and simplicity.

But I can change this part if we'd prefer something like this (from my understanding):
``` Python
i, ucs4lst = 0, []
while i < len(ustr):
  usc4lst.append(usrt[i])
  if LOW < ustr[i] <HIGH:
    usc4lst[-1] += ustr[i]
    i += 1
  i += 1
```

Reply:
You know what? I don't think its worth putting a lot of effort in. Whatever is cleanest.

We don't support 3.3, and if its just for Python 2 support - well, that's also rapidly approaching EOL

Reply:
Agreed, we probably want to polish related code a little bit after ditching Python 2